allure-testng-adaptor/src/main/java/ru/yandex/qatools/allure/testng/AllureTestListener.java

```diff
@@ -124,6 +134,29 @@ public void onTestStart(ITestResult iTestResult) {
         getLifecycle().fire(event);
     }
 
+    private boolean isAtLeastJava8() {
+        try {
+            Class.forName("java.lang.reflect.Parameter");
+            return true;
+        } catch (ClassNotFoundException e) {
+            return false;
+        }
+    }
+
+    private String[] getParameterNames(Method testMethod) {
+        try {
+            Object[] parameters = (Object[]) testMethod.getClass().getMethod("getParameters").invoke(testMethod);
+            String[] result = new String[parameters.length];
+            for (int i = 0; i < parameters.length; i++) {
+                Object parameter = parameters[i];
+                result[i] = (String) parameter.getClass().getMethod("getName").invoke(parameter);
+            }
+            return result;
+        } catch (IllegalAccessException | InvocationTargetException | NoSuchMethodException e) {
```

Comment:
I worry about `ClassCastException`, `IllegalArgumentException`, `NullPointerException`, `ExceptionInInitializerError` etc. Maybe better to catch `Exception`?